packages/snaps-execution-environments/src/common/endowments/index.ts

```diff
@@ -73,9 +78,15 @@ export function createEndowments(
           // This may not have an actual use case, but, safety first.
 
           // We just confirmed that endowmentFactories has the specified key.
-          const { teardownFunction, ...endowment } =
+          let factoryResult;
+          if (EndowmentsWithFactoryOptions.has(endowmentName)) {
             // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-            endowmentFactories.get(endowmentName)!();
+            factoryResult = endowmentFactories.get(endowmentName)!({ snapId });
+          } else {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            factoryResult = endowmentFactories.get(endowmentName)!();
+          }
+          const { teardownFunction, ...endowment } = factoryResult;
```

Comment:
To simplify this logic, how about we just always pass `snapId` to the endowment factory?